### DISCUSSION TITLE
Target not tracking local console drops

### DISCUSSION THREAD

**Discussion by YourStaddy** [jef/streetmerchant#2677] [Help]
Hi I’m very new to this and I am using this product to only monitor target but I have noticed that when consoles go in stock at stores that aren’t set as my default store it does not notify me, I was just wondering if it is something I’m doing wrong(which it probably is) or how to fix it, thank you

**Comment by Mr-Tech-13** [comment]
add this but don't use a proxy or vpn

export const Target: Store = {
  setupAction: async browser => {
    const page = await browser.newPage();
    await page.goto('https://target.com');
    const cookies = await page.cookies();
    const location = cookies.find(cookie => cookie.name === 'fiatsCookie');
    if (location) {
      logger.info('setting target home store cookie');
      await page.setCookie({
        ...location,
        value: '<PUT TARGET COOKIE HERE>',
      });
    }
  }

**Reply by Mr-Tech-13** [reply]
to target.ts

**Reply by YourStaddy** [reply]
Thank you so much for the help but I must be honest that is very confusing to me so would I just copy and paste that word for word, space for space, line for line in the dotenv file or do I need to add more stuff to it? And also what to do with the "to target.ts". Once again thank you so much for taking the time to respond and I apologize for my lack of knowledge and understanding.

**Reply by Mr-Tech-13** [reply]
This is what target.ts should look like. Its under src/store/model/target.ts


import { logger } from '../../logger';
import {Store} from './store';

export const Target: Store = {
  setupAction: async browser => {
    const page = await browser.newPage();
    await page.goto('https://target.com');
    const cookies = await page.cookies();
    const location = cookies.find(cookie => cookie.name === 'fiatsCookie');
    if (location) {
      logger.info('setting target home store cookie');
      await page.setCookie({
        ...location,
        value: '<PUT TARGET COOKIE HERE>',
      });
    }
  },


  currency: '$',
  labels: {
    inStock: [
      {
        container: '[data-test="preorderButton"]',
        text: ['Preorder now'],
      },
      {
        container: '[data-test="shipItButton"]',
        text: ['Ship it'],
      },
    ],
    maxPrice: {
      container: '[data-test="product-price"]',
    },
  },
  links: [
    {
      brand: 'sony',
      model: 'ps5 console',
      series: 'sonyps5c',
      url: 'https://www.target.com/p/playstation-5-console/-/A-81114595',
    },
    {
      brand: 'sony',
      model: 'ps5 digital',
      series: 'sonyps5de',
      url:
        'https://www.target.com/p/playstation-5-digital-edition-console/-/A-81114596',
    },
    {
      brand: 'microsoft',
      model: 'xbox series x',
      series: 'xboxsx',
      url: 'https://www.target.com/p/xbox-series-x-console/-/A-80790841',
    },
    {
      brand: 'microsoft',
      model: 'xbox series s',
      series: 'xboxss',
      url: 'https://www.target.com/p/xbox-series-s-console/-/A-80790842',
    },
  ],
  name: 'target',
};
